hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -250,6 +251,25 @@ public void testTableOperations(HoodieTableType tableType, boolean enableFullSca
     validateMetadata(testTable, emptyList(), true);
   }
 
+  @Test
+  public void testMetadataConcurrentWriters() throws Exception {
+    init(HoodieTableType.MERGE_ON_READ);
+    doWriteOperation(testTable, "0000001", INSERT);
+    AtomicInteger commitTime = new AtomicInteger(2);
+    int i = 1;
+    for (; i <= 50; i++) {
```

Comment:
guess with the fix, we can simplify the test. Whenever metadata table is eligible for archival, we can trigger a table service and ensure archival does not kick in. And following this, we can trigger a regular write and ensure archival kicks in. Anyways, existing test is not very deterministic.